pkg/queue/concurrency_state.go

```diff
@@ -93,33 +95,42 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 	}
 }
 
-// concurrencyStateRequest sends a request to the concurrency state endpoint.
-func concurrencyStateRequest(endpoint string, action string) func() error {
-	return func() error {
-		bodyText := fmt.Sprintf(`{ "action": %q }`, action)
-		body := bytes.NewBufferString(bodyText)
-		req, err := http.NewRequest(http.MethodPost, endpoint, body)
-		if err != nil {
-			return fmt.Errorf("unable to create request: %w", err)
-		}
-		req.Header.Add("Token", "nil") // TODO: use serviceaccountToken from projected volume
-		resp, err := http.DefaultClient.Do(req)
-		if err != nil {
-			return fmt.Errorf("unable to post request: %w", err)
-		}
-		if resp.StatusCode != http.StatusOK {
-			return fmt.Errorf("expected 200 response, got: %d: %s", resp.StatusCode, resp.Status)
-		}
-		return nil
-	}
+type ConcurrencyEndpoint struct {
+	Endpoint  string
+	MountPath string
+	token     atomic.Value
 }
 
-// Pause sends a pause request to the concurrency state endpoint.
-func Pause(endpoint string) func() error {
-	return concurrencyStateRequest(endpoint, "pause")
+func (c ConcurrencyEndpoint) Pause() error { return c.Request("pause") }
+
+func (c ConcurrencyEndpoint) Resume() error { return c.Request("resume") }
+
+func (c ConcurrencyEndpoint) Request(action string) error {
+	bodyText := fmt.Sprintf(`{ "action": %q }`, action)
+	body := bytes.NewBufferString(bodyText)
+	req, err := http.NewRequest(http.MethodPost, c.Endpoint, body)
+	if err != nil {
+		return fmt.Errorf("unable to create request: %w", err)
+	}
+	c.RefreshToken()
```

Comment:
this seems like we're refreshing the token on every request rather than on a timer, but if so we don't _really_ need the atomic, and RefreshToken doesn't need to be exported? Maybe worth converting this to something like:

```golang
ce := NewConcurrencyEndpoint(endpoint, tokenPath)
go func() {
  for range time.NewTicker(1 * time.Minute).C {
    ce.RefreshToken()
  }
}()
```

(which would also let us add a test for RefreshToken, which would be nice).